server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java

```diff
@@ -418,9 +418,6 @@ public static void deleteTable(TableId tableId, boolean insertDeletes, ServerCon
 
     TabletMetadata tablet = context.getAmple().readTablet(extent, FILES, LOGS, PREV_ROW, DIR);
 
-    if (!tablet.getExtent().equals(extent))
```

Comment:
could keep this code, do something like

```java
if(tablet == null){
   throw new RuntimeException("Tablet not "+extent+" not found in metadata ");
}
```

The nice thing about this is there is an error message w/ the extent in it.  Otherwise will just get an NPE later.

Reply:
Added this in 946b7bb